src/CalculatorUITestFramework/HistoryPanel.cs

```diff
@@ -34,22 +49,95 @@ public ReadOnlyCollection<AppiumWebElement> GetAllHistoryListViewItems()
         /// </summary>
         public void ClearHistory()
         {
-            this.HistoryLabel.Click();
 
+            this.HistoryLabel.Click();
             try
             {
-                this.ClearHistoryButton.Click();
+                string sourceB = this.session.PageSource;
```

Comment:
>sourceB [](start = 23, length = 7)

what naming convention are you using here?  What does the "B" stand for?

Reply:
B does not stand for anything.  It was being used to make the variable unique.

Reply:
I'd suggest not doing that and leaving them all as source.  You only need to keep variables unique to a scope.

Reply:
when I could, I'll leave it at source, in the cases were I can't, I'll use a numbering system to make it more clear.
E.g. 
source0
source1